### ISSUE TITLE
Chat and Edit buttons overlap code when selected

### ISSUE THREAD

**Issue by undici77** [continuedev/continue#3373]
### Before submitting your bug report

- [ ] I believe this is a bug. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [ ] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue) that reports the same bug
- [ ] I've seen the [troubleshooting guide](https://docs.continue.dev/troubleshooting) on the Continue Docs

### Relevant environment info

```Markdown
- OS:macOS
- Continue version: v0.9.4
- IDE version: CLion 2024.3.1
- Model: ollama
```

### Description

Chat and Edit buttons overlap code : see screenshots before and after selection!

<img width="714" alt="1" src="https://github.com/user-attachments/assets/092099ac-8726-40a5-89f1-ea37b597e45d" />
<img width="714" alt="2" src="https://github.com/user-attachments/assets/c57e5297-85aa-4de6-9544-cefc33c6f5d2" />

**Comment by undici77** [comment]
Meanwhile, any way to disable these button in CLion?

**Comment by Patrick-Erichsen** [comment]
Hmm, we have logic that should place the tooltip with a 40px margin from the end of a code line, so not sure what is going on here. I'm having trouble reproducing in IntelliJ.

You can disable by going to `Tools | Continue | Display Editor Tooltip`.

**Comment by undici77** [comment]
Maybe a font configuration? Mine (CLion) is

![image](https://github.com/user-attachments/assets/16e4482d-fada-495d-8202-44d3a208a27f)

Let me know if I can help you in some way!

**Comment by Patrick-Erichsen** [comment]
I don't think it is font related because we get the location based on the end of the line: https://github.com/continuedev/continue/blob/main/extensions/intellij/src/main/kotlin/com/github/continuedev/continueintellijextension/listeners/ContinuePluginSelectionListener.kt#L193

One other question, is this behavior pretty consistent for you?

**Comment by undici77** [comment]
Not so much but I can provide you a source code example:

```
struct VeryLongClassNameToTest
{
		VeryLongClassNameToTest(const char *name) : _Name(name)
		{
		}

		virtual ~VeryLongClassNameToTest(void)
		{
			if (_Name != nullptr)
			{
				std::cout << "VeryLongClassNameToTest destructor called test " << std::endl;
			}
		}

		const char *_Name;
};

int main(int argc, const char **argv)
{
	VeryLongClassNameToTest a("test");

	return 0;
}
```

and relative screenshot

![image](https://github.com/user-attachments/assets/8dc992dd-fa48-4a05-8750-15a3df8a1e07)

and I found out that changing runtime the zoom/font size the button don't follow the line position (watch the screenshot)

![image](https://github.com/user-attachments/assets/f6726861-e51a-49e4-9433-1561b4421f60)

Can help?

**Comment by undici77** [comment]
Maybe I found out the issue: I indent code with TAB set to 4. (you can verify my code snippet). If i change TAB with 4 spaces, the position is correct!

With SPACES

![image](https://github.com/user-attachments/assets/5d10e9b0-dfb3-4db8-b674-4b9e838dd212)

With TABS

![image](https://github.com/user-attachments/assets/98cfbe5f-d530-4c78-a35a-bb697f94870d)

**Comment by Patrick-Erichsen** [comment]
Excellent catch! We were calculating position based off of text length, which treats a tab as a single character. Luckily IntelliJ has some simple methods to calculate visual distance.

Should be fixed here: https://github.com/continuedev/continue/pull/3717

**Comment by danielschilling-ml** [comment]
I am experiencing this same issue in Rider. We also use tabs for indentation. I'm so glad that there is a fix on the way! I assume the same fix will also apply to Rider?

![Image](https://github.com/user-attachments/assets/dc972ab8-de1f-4678-8cd0-e8beffd9f9e2)

**Comment by danielschilling-ml** [comment]
I just downloaded and installed 0.0.86, and it does fix the problem in Rider! Thank you!

![Image](https://github.com/user-attachments/assets/c7a2db1b-b398-49fd-b0b1-46a6aae39ded)

**Comment by RomneyDa** [comment]
Thanks all!
